Sources/Kitura/contentType/MediaType.swift

```diff
@@ -0,0 +1,147 @@
+        self.topLevelType = type
+        self.subtype = subtype.lowercased()
+    }
+    
+    /**
+     Initialize a `MediaType` instance from a `String`. If no subtype is provided it will default to "*" representing all subtypes.
+     If the string preceding the first "/" is not a valid `TopLevelType` the initializer will return nil.
+     ### Usage Example: ###
+     ```swift
+     let mediaType = MediaType("application/json")
+     ```
+     */
+    public init? (_ mimeType: String) {
+        let mimeComponents = mimeType
+            .lowercased()
+            .components(separatedBy: "/")
+        guard let topLevelType = TopLevelType(rawValue: mimeComponents[0]) else {
+            return nil
+        }
+        self.topLevelType = topLevelType
+        if mimeComponents.indices.contains(1) && !mimeComponents[1].isEmpty {
+            self.subtype = mimeComponents[1]
+        } else {
+            self.subtype = "*"
+        }
+    }
+    
+    /**
+     Initialize a `MediaType` instance from a `KituraNet` `HeadersContainer`. This will extract the media type String from the first section of the "Content-type" header and use this to initialize itself. If the string preceding the first "/" is not a valid `TopLevelType` the initializer will return nil.
+     ### Usage Example: ###
+     ```swift
+     let headers = HeadersContainer()
+     headers.append("Content-Type", value: ["application/json"])
+     let mediaType = MediaType(headers: headers)
+     ```
+     */
+    public init? (headers: HeadersContainer) {
+        let contentType = headers["Content-Type"]?[0]
+        guard let contentTypeComponents = contentType?.components(separatedBy: ";") else {
+            return nil
+        }
+        self.init(contentTypeComponents[0])
+    }
+    
+    // MARK: Helper Constructors
+    
+    /**
+     Helper constructor for the "application/json" media type
+     ### Usage Example: ###
+     ```swift
+     let mediaType = MediaType.json
+     print(mediaType.description)
+     // Prints ("application/json")
+     ```
+     */
+    public static let json = MediaType(type: .application, subtype: "json")
+    
+    /**
+     Helper constructor for the "application/x-www-form-urlencoded" media type
+     ### Usage Example: ###
+     ```swift
+     let mediaType = MediaType.urlEncoded
+     print(mediaType.description)
+     // Prints ("application/x-www-form-urlencoded")
+     ```
+     */
+    public static let urlEncoded = MediaType(type: .application, subtype: "x-www-form-urlencoded")
+    
+    // MARK: Protocol conformance
+    
+    /**
+     Returns the media type, as a String structured: `topLevelType`/`subtype`. Required for CustomStringConvertible conformance.
+     ### Usage Example: ###
+     ```swift
+     print(mediaType.description)
+     // Prints ("application/json")
+     ```
+     */
+    public var description: String {
+        return "\(topLevelType)/\(subtype)"
```

Comment:
This will recalculate the description every time the property is accessed. It would be better to use `didSet{}` on topLevelType and subtype to recalculate the String only when necessary.